tests/components/insteon/test_config_flow.py

```diff
@@ -0,0 +1,703 @@
+    assert len(config_entry.options[CONF_OVERRIDE]) == 1
+    assert config_entry.options[CONF_OVERRIDE][0][CONF_ADDRESS] == "1A.2B.3C"
+    assert config_entry.options[CONF_OVERRIDE][0][CONF_CAT] == 4
+    assert config_entry.options[CONF_OVERRIDE][0][CONF_SUBCAT] == 170
+
+    result2 = await _options_init_form(hass, config_entry.entry_id, STEP_ADD_OVERRIDE)
+
+    user_input = {
+        CONF_ADDRESS: "4d5e6f",
+        CONF_CAT: "05",
+        CONF_SUBCAT: "bb",
+    }
+    await _options_form(hass, result2["flow_id"], user_input)
+
+    assert len(config_entry.options[CONF_OVERRIDE]) == 2
+    assert config_entry.options[CONF_OVERRIDE][1][CONF_ADDRESS] == "4D.5E.6F"
+    assert config_entry.options[CONF_OVERRIDE][1][CONF_CAT] == 5
+    assert config_entry.options[CONF_OVERRIDE][1][CONF_SUBCAT] == 187
+
+
+async def test_options_remove_device_override(hass: HomeAssistantType):
+    """Test removing a device override."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        entry_id="abcde12345",
+        data={**MOCK_USER_INPUT_HUB_V2, CONF_HUB_VERSION: 2},
+        options={
+            CONF_OVERRIDE: [
+                {CONF_ADDRESS: "1A.2B.3C", CONF_CAT: 6, CONF_SUBCAT: 100},
+                {CONF_ADDRESS: "4D.5E.6F", CONF_CAT: 7, CONF_SUBCAT: 200},
+            ]
+        },
+    )
+
+    config_entry.add_to_hass(hass)
+    result = await _options_init_form(hass, config_entry.entry_id, STEP_REMOVE_OVERRIDE)
+
+    user_input = {CONF_ADDRESS: "1A.2B.3C"}
+    result, _ = await _options_form(hass, result["flow_id"], user_input)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert len(config_entry.options[CONF_OVERRIDE]) == 1
+
+
+async def test_options_remove_device_override_with_x10(hass: HomeAssistantType):
+    """Test removing a device override when an X10 device is configured."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        entry_id="abcde12345",
+        data={**MOCK_USER_INPUT_HUB_V2, CONF_HUB_VERSION: 2},
+        options={
+            CONF_OVERRIDE: [
+                {CONF_ADDRESS: "1A.2B.3C", CONF_CAT: 6, CONF_SUBCAT: 100},
+                {CONF_ADDRESS: "4D.5E.6F", CONF_CAT: 7, CONF_SUBCAT: 200},
+            ],
+            CONF_X10: [
+                {
+                    CONF_HOUSECODE: "d",
+                    CONF_UNITCODE: 5,
+                    CONF_PLATFORM: "light",
+                    CONF_DIM_STEPS: 22,
+                }
+            ],
+        },
+    )
+
+    config_entry.add_to_hass(hass)
+    result = await _options_init_form(hass, config_entry.entry_id, STEP_REMOVE_OVERRIDE)
+
+    user_input = {CONF_ADDRESS: "1A.2B.3C"}
+    result, _ = await _options_form(hass, result["flow_id"], user_input)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert len(config_entry.options[CONF_OVERRIDE]) == 1
+    assert len(config_entry.options[CONF_X10]) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)